### DISCUSSION TITLE
Demucs integration for desktop?

### DISCUSSION THREAD

**Discussion by Myzel394** [KRTirtho/spotube#408] [Ideas]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Category of the Feature

Logical Improvements

### Credits

Me? Facebook? Not sure what to put in here

### Pain

I don't want to listen to lyrics. I sometimes only want to listen to the instrumentals.

### Suggested Solution

Demucs (https://github.com/facebookresearch/demucs) is an AI that allows you to create near-perfect instrumentals from any song.  If we embed the Python interpreter into Spotube, we could offer the functionality to create instrumentals from songs using an AI directly via the GUI - how cool is that?!? Imagine all you need to do is just to press a button and your computer will literally use an AI to generate an instrumental version of the song you are currently listening to :exploding_head: The only different part will be to embed the Python interpreter, but afaik there have been several projects that have done this before, maybe we could learn from then on how to do that

### Useful resources/references

* Here's the whitepaper (you don't need to understand it, I do not understand it at all, you just need to copy the commands): https://github.com/facebookresearch/demucs
* There's a GUI for Windows that allows you to extract songs using AIs offline: https://github.com/Anjok07/ultimatevocalremovergui

**Comment by KRTirtho** [comment]
> If we embed the Python interpreter into Spotube

Spotube size would be 50MB+. And just for a single feature, it's not justifiable
The best approach would be using an API that takes the YouTube URL/Track Bytes and creates separate tracks + caches
